sync/awsdatabricks.py

```diff
@@ -886,91 +913,111 @@ def _cluster_log_destination(
 
 def monitor_cluster(cluster_id: str, polling_period: int = 30) -> None:
     cluster = get_default_client().get_cluster(cluster_id)
-    cluster_log_dest = _cluster_log_destination(cluster)
-    if cluster_log_dest:
-        (_, filesystem, bucket, base_prefix) = cluster_log_dest
-        # If the event log destination is just a *bucket* without any sub-path, then we don't want to include
-        #  a leading `/` in our Prefix (which will make it so that we never actually find the event log), so
-        #  we make sure to re-strip our final Prefix
-        file_key = f"{base_prefix}/sync_data/cluster_instances.json".strip("/")
+    spark_context_id = cluster.get("spark_context_id")
 
-        aws_region_name = DB_CONFIG.aws_region_name
-        ec2 = boto.client("ec2", region_name=aws_region_name)
+    while not spark_context_id:
+        # This is largely just a convenience for when this command is run by someone locally
+        logger.info("Waiting for cluster startup...")
+        sleep(30)
+        cluster = get_default_client().get_cluster(cluster_id)
+        spark_context_id = cluster.get("spark_context_id")
 
-        if filesystem == "s3":
-            s3 = boto.client("s3")
+    (log_url, filesystem, bucket, base_prefix) = _cluster_log_destination(cluster)
+    if log_url:
+        _monitor_cluster(
+            (log_url, filesystem, bucket, base_prefix), cluster_id, spark_context_id, polling_period
+        )
+    else:
+        logger.warning("Unable to monitor cluster due to missing cluster log destination - exiting")
 
-            def write_file(body: bytes):
-                s3.put_object(Bucket=bucket, Key=file_key, Body=body)
 
-        elif filesystem == "dbfs":
+def _monitor_cluster(
+    cluster_log_destination, cluster_id: str, spark_context_id: int, polling_period: int
+) -> None:
+    (log_url, filesystem, bucket, base_prefix) = cluster_log_destination
```

Comment:
It'd be nice to define this at some point. We could tack on `cluster_id` to the `ParseResult` path or create our own named tuple-based type.